src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs

```diff
@@ -263,6 +265,12 @@ private CallbackAppTraceSource CreateTracer(AgentTaskPluginExecutionContext cont
             var tracer = new CallbackAppTraceSource(str => context.Output(str), System.Diagnostics.SourceLevels.Information);
             return tracer;
         }
+
+        private static long GetArtifactSize(string path)
+        {
+            DirectoryInfo directory = new DirectoryInfo(path);
+            return directory.EnumerateFiles("*", SearchOption.AllDirectories).Select(x => x.Length).Sum();
```

Comment:
LGTM. Wonder what will be the perf impact for large number of files.

Reply:
I think it's a performance hit we have to take somewhere for us to have this info. For file container artifacts, I was able to modify the upload code to return the file size uploaded. It didn't look like we had the same option for Pipeline Artifacts as that's not returned by the blob upload client (and it doesn't seem to be owned by any of our teams).

According to the docs: https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.enumeratefiles?view=netframework-4.8
It is somewhat efficient in that it returns an enumerable that we can begin using rather than waiting for the full result back first.

Reply:
Sounds good. A quick thing, did you test this with just uploading/downloading the empty directory?

Reply:
Yep, just verified it works with an empty directory as expected and the artifact size for it is 0.

Reply:
Make sure Length is a `long` and that `Select<long>` and that `Sum<long>`.  Sometimes one of the three will be an `int` and you'll overflow in weird ways.

Reply:
Hold up.  We are already computing the total size.  It's the `TransitiveContentSize`.  Just need to put it in the `PublishResult`

Reply:
Oh, that's great! Let me know when that's added and I'll update this PR (I'm assuming the version of the Blobstore WebApi will have to be bumped)

Reply:
👍